### PR TITLE
Fix gitmodules conflicts

### PR DESCRIPTION
**Pull request by bpeabody**
depends on https://github.com/twosigma/git-meta/pull/146

### CODE REVIEW

File: node/lib/util/read_repo_ast_util.js [hunk 1]
```diff
@@ -47,6 +47,24 @@ const RebaseFileUtil      = require("./rebase_file_util");
...
+ * Load the submodules objects from the specified `repo` on the specified
+ * `commitId`.
+ * @param {NodeGit.Repository} repo
+ * @param {Object}             urls
+ * @param {NodeGit.Commit}     commit
+ */
+const getSubmodules = co.wrap(function *(repo, urls, commit) {
+    let result = {};
+    const tree = yield commit.getTree();
+    for (let key in urls) {
```
Comment (novalis) [inline node/lib/util/read_repo_ast_util.js]: rename var "key" to "url"?

File: node/lib/util/submodule_config_util.js [hunk 2]
```diff
@@ -403,3 +403,101 @@ exports.initSubmoduleAndRepo = co.wrap(function *(repoUrl,
...
+    }
+    for (let key in rhs) {
+        lhsValue = result[key];  // use 'result' as it may be smaller
+        rhsValue = rhs[key];
+        mergeBaseValue = mergeBase[key];
+
+        // We will have a conflict only when the value doesn't exist in 'lhs'
+        // -- otherwise, it would have been detected already.  So, a conflict
+        // exists when it's gone from the `lhs`, but present in the `rhs`, and
+        // there is a value in `mergeBase` that's different from `lhs`.
```
Comment (novalis) [inline node/lib/util/submodule_config_util.js]: Replace "lhs" with "rhs" in this line.

File: node/lib/util/rebase_util.js [hunk 3]
```diff
@@ -403,6 +439,34 @@ up-to-date.`);
...
+                        const fromNext = yield getSubs(metaRepo, next);
+                        const fromBase = yield getMergeBaseSubs();
+                        const fromOnto = yield getOntoSubs();
+                        const merged = Conf.mergeSubmoduleConfigs(fromNext,
+                                                                  fromOnto,
+                                                                  fromBase);
+                        // If it was resolved, write out and stage the new
+                        // modules state.
+
+                        if (null !== merged) {
```
Comment (novalis) [inline node/lib/util/rebase_util.js]: Should we give a better message here?  Or show the file in state with merge markers?

Comment (bpeabody) [inline node/lib/util/rebase_util.js]: I think that if I do nothing there, libgit2 will have the merge markers in place. Will verify.

Comment (bpeabody) [inline node/lib/util/rebase_util.js]:
```bash
$ git meta rebase foo
There is a conflict in .gitmodules.
$ cat .gitmodules
[submodule "s"]
        path = s
        url = /Users/peabody/trash/rebase/s
[submodule "q"]
        path = q
<<<<<<< ab1db74303de44c298af005599f3c29171718936
        url = /Users/peabody/trash/rebase/r
=======
        url = /Users/peabody/trash/rebase/q

**Comment by bpeabody** [comment]
OK, I think I've got everything.

**Pull request merged**